### PR TITLE
week 5 solved repdigit with rspec tests

### PR DESCRIPTION
**Pull request by neb5562**

### CODE REVIEW

File: week5/repdigit/spec/bootstrap_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+100.times do
+  describe RepDigit do
+    # must raise AttributeError fo ar negative numbers as it will not pass validation
+    it "raises AttributeError" do
+      expect{RepDigit.new(Faker::Number.negative).call}.to raise_error(AttributeError)
+    end
+
+    # will generate random numbers like 3333333, 555555555555555, 22222222. all msut be true
+    it "returns true" do
+      expect(RepDigit.new(Faker::Number.within(range: NUMBER_CHAR_FROM..NUMBER_CHAR_TO).to_s * rand(SYMBOLS_NUMBER_FROM...SYMBOLS_NUMBER_TO).to_i).call).to be_truthy
```
Comment (neb5562) [inline week5/repdigit/spec/bootstrap_spec.rb]: randomly generate numbers for test

File: week5/repdigit/lib/random_number_helper.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module RandNumHelper
+  def self.non_repdigit_number
+    left = rand(1..5)
+    right = rand(5..9)
+    middle = rand(99..9999)
+
+    (left.to_s + right.to_s + middle.to_s).to_i
```
Comment (neb5562) [inline week5/repdigit/lib/random_number_helper.rb:7]: this helper will generate **non repdigit** numbers!

File: week5/repdigit/lib/repdigit.rb [hunk 3]
```diff
@@ -11,8 +14,9 @@ def initialize(digit)
   def call
     validate_digit
     repdigit?
-  #rescue AttributeError => e
-   # warn e
+  rescue AttributeError => e
+   warn e
+   return false
```
Comment (neb5562) [inline week5/repdigit/lib/repdigit.rb:19]: warn message will show in testing progress but with return false tests will pass

File: week5/repdigit/spec/bootstrap_spec.rb [hunk 4]
```diff
@@ -30,7 +30,9 @@
 
     # will generate random numbers like 3333333, 555555555555555, 22222222. all msut be true
     it "returns true" do
-      expect(RepDigit.new(Faker::Number.within(range: NUMBER_CHAR_FROM..NUMBER_CHAR_TO).to_s * rand(SYMBOLS_NUMBER_FROM...SYMBOLS_NUMBER_TO).to_i).call).to be_truthy
+      symbol = Faker::Number.within(range: NUMBER_CHAR_FROM..NUMBER_CHAR_TO).to_s
```
Comment (neb5562) [inline week5/repdigit/spec/bootstrap_spec.rb:33]: fixed long line

File: week5/repdigit/lib/repdigit.rb [hunk 5]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env ruby
-$LOAD_PATH << './lib'
+$LOAD_PATH << '../lib'
```
Comment (neb5562) [inline week5/repdigit/lib/repdigit.rb:2]: now works every time you call this file directly or require it in another file

File: week5/repdigit/spec/bootstrap_spec.rb (hunk 1 above)
Comment (alexshvedko) [reply week5/repdigit/spec/bootstrap_spec.rb]: ok

File: week5/repdigit/lib/repdigit.rb (hunk 5 above)
Comment (alexshvedko) [reply week5/repdigit/lib/repdigit.rb:2]: ok

**Review by alexshvedko — APPROVED** [review]

**Pull request merged**